src/embed/hostEventClient/host-event-client.ts

```diff
@@ -0,0 +1,82 @@
+import { HostEvent } from '../../types';
+import { processTrigger } from '../../utils/processTrigger';
+import {
+    UiPassthroughArrayResponse,
+    UiPassthroughEvent, HostEventRequest, HostEventResponse,
+    UiPassthroughRequest,
+    UiPassthroughResponse,
+} from './contracts';
+
+export class HostEventClient {
+  thoughtSpotHost: string;
+
+  constructor(thoughtSpotHost: string) {
+      this.thoughtSpotHost = thoughtSpotHost;
+  }
+
+  public async executeUiPassthroughApi<UiPassthroughEventT extends UiPassthroughEvent>(
+      iFrame: HTMLIFrameElement,
+      apiName: UiPassthroughEventT,
+      parameters: UiPassthroughRequest<UiPassthroughEventT>,
+  ): UiPassthroughArrayResponse<UiPassthroughEventT> {
+      const res = await processTrigger(iFrame, HostEvent.UiPassthrough, this.thoughtSpotHost, {
+          type: apiName,
+          parameters,
+      });
+
+      return res;
+  }
+
+  public async handleUiPassthroughForHostEvent<UiPassthroughEventT extends UiPassthroughEvent>(
```

Comment:
Let's name these on the use-case of it instead. Eg: handleHostEventWithParams

Reply:
done